monkey_island/cc/services/report.py

```diff
@@ -0,0 +1,422 @@
+import ipaddress
+
+from cc.database import mongo
+from cc.services.config import ConfigService
+from cc.services.edge import EdgeService
+from cc.services.node import NodeService
+from cc.utils import local_ip_addresses, get_subnets
+
+__author__ = "itay.mizeretz"
+
+
+class ReportService:
+    def __init__(self):
+        pass
+
+    EXPLOIT_DISPLAY_DICT = \
+        {
+            'SmbExploiter': 'SMB Exploiter',
+            'WmiExploiter': 'WMI Exploiter',
+            'SSHExploiter': 'SSH Exploiter',
+            'RdpExploiter': 'RDP Exploiter',
+            'SambaCryExploiter': 'SambaCry Exploiter',
+            'ElasticGroovyExploiter': 'Elastic Groovy Exploiter',
+            'Ms08_067_Exploiter': 'Conficker Exploiter',
+            'ShellShockExploiter': 'ShellShock Exploiter',
+        }
+
+    class ISSUES_DICT:
+        WEAK_PASSWORD = 0
+        STOLEN_CREDS = 1
+        ELASTIC = 2
+        SAMBACRY = 3
+        SHELLSHOCK = 4
+        CONFICKER = 5
+
+    class WARNINGS_DICT:
+        CROSS_SEGMENT = 0
+        TUNNEL = 1
+
+    @staticmethod
+    def get_first_monkey_time():
+        return mongo.db.telemetry.find({}, {'timestamp': 1}).sort([('$natural', 1)]).limit(1)[0]['timestamp']
+
+    @staticmethod
+    def get_last_monkey_dead_time():
+        return mongo.db.telemetry.find({}, {'timestamp': 1}).sort([('$natural', -1)]).limit(1)[0]['timestamp']
+
+    @staticmethod
+    def get_monkey_duration():
+        delta = ReportService.get_last_monkey_dead_time() - ReportService.get_first_monkey_time()
```

Comment:
There has to be a better way to parse datetime then this...  You can probably use a strftime statement.

Reply:
Simplified it as much as I could.
There's no built-in function which parses it
https://stackoverflow.com/questions/8906926/formatting-python-timedelta-objects